services/cd-service/pkg/repository/repository.go

```diff
@@ -914,7 +914,11 @@ func (s *State) GetEnvironmentConfigsAndValidate(ctx context.Context, transactio
 	if len(envConfigs) == 0 {
 		logger.Warn("No environment configurations found. Check git settings like the branch name. Kuberpult cannot operate without environments.")
 	}
+
 	for envName, env := range envConfigs {
+		if env.ArgoCdConfigs != nil && env.ArgoCd != nil {
+			logger.Warn(fmt.Sprintf("The environment '%s' has both ArgoCdConfigs and ArgoCd configured. This is not supported", envName))
```

Comment:
```suggestion
			logger.Error(fmt.Sprintf("The environment '%s' has both ArgoCdConfigs and ArgoCd configured. This is not supported", envName))
```
I think this is critical enough to be worthy an error.